<h1 style='font-family: Garamond ;'>Домашнє завдання до теми "Основи роботи з рядками та математичні операції"</h1>
<hr></hr>

<ul style='font-family: Garamond ; font-size: 16px;'>
    <li>1. Створіть змінні <code>a</code>, <code>b</code>, <code>c</code> та <code>d</code>, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:</li>.
</ul>

$$ \frac{(a+b) * c}{d} $$

<p style='font-family: Garamond ; font-size: 16px;'><b>Важливо</b>: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.</p>



In [14]:
def calculation(a, b, c, d):
    '''
    Calculates the result using the formula ((a + b) * c) / d.

    Parameters:
    ----------
    a : int or float
        The first addend for the addition operation.
    b : int or float
        The second addend for the addition operation.
    c : int or float
        The multiplier for the sum of a and b.
    d : int or float
        The divisor for the division operation.

    Returns:
    -------
    float
        The quotient of the division ((a + b) * c) / d, if d ≠ 0.
    
    Notes:
    ------
    - If `d` equals 0, the function prints an error message 
      "Divisor can not be a zero, otherwise your answer is infinity!" 
      and does not return any value.
    
    Example:
    --------
    >>> calculation(2, 3, 4, 5)
    4.0

    >>> calculation(2, 3, 4, 0)
    Divisor can not be a zero, otherwise your answer is infinity!

    '''
    
    if d == 0:
        print('Divisor can not be a zero, otherwise your answer is infinity!')
    else:    
        sum = a + b
        dividend = sum * c
        quotient = dividend / d
        return quotient
a = 101
b = 210
c = 111
d = 402
result = calculation(a, b, c, d)
print(result)

85.8731343283582


<p style='font-family: Garamond ; font-size: 16px;'>Очікувана відповідь: 85.8731343283582 .</p>

<ul style='font-family: Garamond ; font-size: 16px;'>
    <li>2. Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.</li>
</ul>

In [3]:
print(11283 % 89)

69


<ul style='font-family: Garamond ; font-size: 16px;'>
    <li>3. Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)</li>
</ul>

In [4]:
print(11283 // 89)

126


<ul style='font-family: Garamond ; font-size: 16px;'>
    <li>4. Запишіть у змінну <code>name</code> ваше імʼя.</li>
</ul>

In [10]:
name = 'Veronika'

<ul style='font-family: Garamond ; font-size: 16px;'>
    <li>5. Запишіть у змінну <code>surname</code> ваше прізвище.</li>
</ul>

In [6]:
surname = 'Kobets'

<ul style='font-family: Garamond ; font-size: 16px;'>
    <li>6. Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну <code>name_surname</code> та виведіть результат на екран.</li>
</ul>

In [15]:
name_surname = name + ' ' + surname
name_surname

'Veronika Kobets'

<p style='font-family: Garamond ; font-size: 16px;'>Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva'</p>

<ul style='font-family: Garamond ; font-size: 16px;'>
    <li>7. Зробіть усі літери змінної <code>name_surname</code> заглавними використовуючи метод рядків в Python:</li>
</ul>

In [16]:
name_surname.upper()

'VERONIKA KOBETS'

<p style='font-family: Garamond ; font-size: 16px;'>Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'</p>

<ul style='font-family: Garamond ; font-size: 16px;'>
    <li>8. Виведіть третю літеру вашого <code>name_surname</code>. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.</li>
</ul>

In [18]:
name_surname[2]

'r'

<p style='font-family: Garamond ; font-size: 16px;'>Очікувана відповідь (тільки з вашими даними): 'n'</p>

<ul style='font-family: Garamond ; font-size: 16px;'>
    <li>9. Виведіть першу та останню літеру вашого прізвища.</li>
</ul>

In [7]:
surname[0], surname[-1]

('K', 's')

<p style='font-family: Garamond ; font-size: 16px;'>Очікувана відповідь (тільки з вашими даними): ('P', 'a')</p>

<ul style='font-family: Garamond ; font-size: 16px;'>
    <li>10. Виведіть довжину вашого прізвища.</li>
</ul>

In [9]:
len(surname)

6

<p style='font-family: Garamond ; font-size: 16px;'>Очікувана відповідь (тільки з вашими даними): 7</p>

<ul style='font-family: Garamond ; font-size: 16px;'>
    <li>11. Виведіть кількість літер <code>a</code> у столиці Мадагаскару <code>Antananarivo</code>. Важливо врахувати всі літери, як великі, так і маленькі.</li>
</ul>

In [11]:
madagascar_capital = 'Antananarivo'
a_counts = madagascar_capital.lower().count('a')
a_counts

4

<p style='font-family: Garamond ; font-size: 16px;'>Очікувана відповідь: 4</p>

<ul style='font-family: Garamond ; font-size: 16px;'>
    <li>12. Створіть рядок <code>my_string</code> зі значенням: "The quick brown fox jumps over the lazy dog!!!".</li>
</ul>
<p style='font-family: Garamond ; font-size: 16px;'>Перевірте, чи починається рядок <code>my_string</code> з "The" і чи закінчується він "dog!". Виведіть результати на екран.</p>

In [12]:
my_string = 'The quick brown fox jumps over the lazy dog!!!'
starts_with_the = my_string.startswith("The")
ends_with_dog = my_string.endswith("dog!")

starts_with_the, ends_with_dog

(True, False)

<ul style='font-family: Garamond ; font-size: 16px;'>
    <li>13. Розділіть рядок <code>my_string</code> на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.</li>
</ul>

In [13]:
my_string = 'The quick brown fox jumps over the lazy dog!!!'
my_string_splitted = my_string.split(' ')
print(my_string_splitted, type(my_string_splitted))

joined_my_string_splitted = '_'.join(my_string_splitted)
print(f'\n{joined_my_string_splitted}')

['The', 'quick', 'brown', 'fox', 'jumps', 'over', 'the', 'lazy', 'dog!!!'] <class 'list'>

The_quick_brown_fox_jumps_over_the_lazy_dog!!!


<p style='font-family: Garamond ; font-size: 16px;'>Очікуваний результат: <code>The_quick_brown_fox_jumps_over_the_lazy_dog!!!</code></p>